### DISCUSSION TITLE
[Help]: How to distinguish click and drag events in Embla

### DISCUSSION THREAD

**Discussion by sarussss** [davidjerleke/embla-carousel#712] [Help] [answered]
### Summary

How to distinguish click and drag events in Embla.
Some event similar to this
https://flickity.metafizzy.co/events#dragend
https://flickity.metafizzy.co/events#staticclick
With Embla I just found the drag detection event but don't know how to find the click detection
https://www.embla-carousel.com/api/events/#pointerup
 Please give me any tips that could make it work. like what should I do.

**Comment by davidjerleke** [comment, marked as answer]
Hi @sarussss,

Thank you for your question. 

> With Embla I just found the drag detection event but don't know how to find the click detection

Embla will stop the event propagation if it decides that the gesture is a drag gesture, meaning that the click event won't fire in these cases. So you should be able to do this:


```ts
emblaApi.slideNodes().forEach((slideNode, slideIndex) => {
  slideNode.addEventListener('click', () => {
    console.log(`You just clicked on slide ${slideIndex + 1}`)
  })
})
```

...and **it should only fire** when the **gesture** is a **click event**, and not a drag event. You can control how sensitive this "click blocker" should be using the [`dragThreshold`](https://www.embla-carousel.com/api/options/#dragthreshold) option.

Let me know if this helps.

Best,
David

**Reply by sarussss** [reply]
Hi David, Thank you for your answer. 
That's what i need.
